_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp

```diff
@@ -2068,28 +2083,34 @@ void ImplementationAvc::BrcPreEnc(
     m_brc.PreEnc(task.m_brcFrameParams, m_tmpVmeData);
 }
 #if defined(MXF_ENABLE_MCTF_IN_AVC)
-mfxStatus ImplementationAvc::SubmitToMctf(DdiTask * pTask, bool isSceneChange, bool doIntraFiltering)
+mfxStatus ImplementationAvc::SubmitToMctf(DdiTask * pTask)
 {
     MFX_AUTO_LTRACE(MFX_TRACE_LEVEL_API, "VideoVPPHW::SubmitToMctf");
-    //mfxStatus sts = MFX_ERR_NONE;
 
-    pTask->m_bFrameReady   = false;
-    mfxFrameSurface1
-        * pSurfI         = nullptr;
-    pSurfI               = m_core->GetNativeSurface(pTask->m_yuv);
-    pSurfI               = pSurfI ? pSurfI : pTask->m_yuv;
-
-    amtMctf->IntBufferUpdate(isSceneChange);
+    pTask->m_bFrameReady = false;
+    bool isIntraFrame    = (pTask->GetFrameType() & MFX_FRAMETYPE_I) || (pTask->GetFrameType() & MFX_FRAMETYPE_IDR);
+    bool isPFrame        = (pTask->GetFrameType() & MFX_FRAMETYPE_P);
+    bool isAnchorFrame   = isIntraFrame || isPFrame;
 
+    amtMctf->IntBufferUpdate(pTask->m_SceneChange, isIntraFrame, pTask->m_doMCTFIntraFiltering);
+    if (isAnchorFrame || pTask->m_SceneChange)
+    {
+        pTask->m_idxMCTF = FindFreeResourceIndex(m_mctf);
+        pTask->m_midMCTF = AcquireResource(m_mctf, pTask->m_idxMCTF);
+        if (!pTask->m_midMCTF)
```

Comment:
Can you try to artificially set number of surfaces =1, step into the case when you need more than 1 surface and check whether we will correctly exit here w/ MFX_TASK_BUSY and continue to work?

Reply:
will do